### PR TITLE
remove rustc-serialize dependency and factor `util::cargo` out into a crate

### PR DESCRIPTION
**Pull request by oli-obk**
I need the `cargo` module in `miri`

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -32,14 +32,15 @@ test = false
...
 
 [dev-dependencies]
 compiletest_rs = "0.2.5"
 lazy_static = "0.1.15"
 regex = "0.2"
-rustc-serialize = "0.3"
+serde_derive = "0.9.0-rc3"
 clippy-mini-macro-test = { version = "0.1", path = "mini-macro" }
-serde = "0.7"
+serde = "0.9.0-rc3"
```
Comment (mcarton) [inline Cargo.toml]: Why a rc?

File: util/lintlib.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 level_re = re.compile(r'''(Forbid|Deny|Warn|Allow)''')
 conf_re = re.compile(r'''define_Conf! {\n([^}]*)\n}''', re.MULTILINE)
 confvar_re = re.compile(
-    r'''/// Lint: (\w+). (.*).*\n *\("([^"]*)", (?:[^,]*), (.*) => (.*)\),''')
+    r'''/// Lint: (\w+). (.*).*\n\s*\("([^"]*)",\s*(?:[^,]*),\s*(.*)\s*=>\s*(.*)\),''', re.MULTILINE)
```
Comment (mcarton) [inline util/lintlib.py]: This escaped #1470 :smile:

**Comment by llogiq** [comment]
LGTM modulo travis

File: Cargo.toml (hunk 1 above)
Comment (oli-obk) [inline Cargo.toml]: because there's no full release yet

Comment (oli-obk) [inline Cargo.toml]: but there is now... updating

**Comment by llogiq** [comment]
@oli-obk Should we merge #1470 first?

**Comment by oli-obk** [comment]
They are independent

**Pull request merged**